x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/search/action/RestDeleteSearchApplicationActionTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.search.action;
+
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.client.NoOpNodeClient;
+import org.elasticsearch.test.rest.FakeRestChannel;
+import org.elasticsearch.test.rest.FakeRestRequest;
+import org.elasticsearch.xcontent.NamedXContentRegistry;
+
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class RestDeleteSearchApplicationActionTests extends ESTestCase {
+    public void testWithNonCompliantLicense() throws Exception {
+        final XPackLicenseState licenseState = mock(XPackLicenseState.class);
+        when(licenseState.getOperationMode()).thenReturn(License.OperationMode.BASIC);
```

Comment:
This mocking for returning BASIC operation mode is not really necessary (here and in other similar tests) since the license checking is done with `XPackLicenseState.isAllowed(...)` which by default returns `false` for a mock object. I think the test would still pass even if this returns `ENTERPRISE`. So I'd just drop this line.